frontend/components/utils/BidInput.vue

```diff
@@ -0,0 +1,137 @@
+                <div v-if="!value" class="absolute text-right right-8 top-1">
+                    <format-currency v-if="totalPrice && !disabled" :value="totalPrice" />
+                    <span v-else class="opacity-50">UNKNOWN</span>
+                </div>
+                <span class="absolute right-1 top-1">DAI</span>
+                <input
+                    v-model="userInput"
+                    class="Input"
+                    :class="{ Error: error }"
+                    :disabled="disabled"
+                    @focus="hideTotalPrice()"
+                    @blur="validateFinished()"
+                />
+            </label>
+        </Tooltip>
+    </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import BigNumber from 'bignumber.js';
+import { Tooltip } from 'ant-design-vue';
+import FormatCurrency from './FormatCurrency.vue';
+
+export default Vue.extend({
+    components: { FormatCurrency, Tooltip },
+    props: {
+        totalPrice: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: null,
+        },
+        minimumDepositDai: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: null,
+        },
+        value: {
+            type: Object as Vue.PropType<BigNumber> | undefined,
+            default: undefined,
+        },
+        disabled: {
+            type: Boolean,
+            default: false,
+        },
+    },
+    data() {
+        return {
+            userInput: undefined as string | undefined,
+            error: undefined as string | undefined,
+        };
+    },
+    computed: {
+        showErrorPopup(): boolean {
+            return !!this.error;
+        },
+    },
+    watch: {
+        value(newVal: BigNumber | undefined): void {
+            if (!newVal) {
+                this.resetToTotal();
+                return;
+            }
+            if (newVal.isEqualTo(this.totalPrice)) {
+                this.resetToTotal();
+            }
+            if (newVal.isEqualTo(this.minimumDepositDai)) {
+                this.userInput = newVal.toString();
+            }
+        },
+        error(newVal): void {
+            this.$emit('error', newVal);
+        },
+        userInput(newVal, oldVal) {
+            this.validateInput(newVal, oldVal);
+        },
+    },
+    methods: {
+        validateInput(newVal: string, oldVal: string): void {
+            this.error = undefined;
+            if (!newVal) {
+                return;
+            }
+            const value = new BigNumber(newVal);
+            if (isNaN(value.toNumber())) {
+                this.userInput = oldVal;
+            } else if (value.isGreaterThan(this.totalPrice)) {
+                this.error = `The bidding amount can not be greater than ${this.totalPrice.toFixed(2)} DAI`;
+            } else if (value.isLessThan(this.minimumDepositDai)) {
+                this.error = `The bidding amount can not be smaller than ${this.minimumDepositDai.toFixed(2)} DAI`;
```

Comment:
Right now, when the error is shown, we still show the `The amount to receive` value:

<img width="606" alt="Screenshot 2022-03-08 at 17 32 52" src="https://user-images.githubusercontent.com/3393626/157282316-bde0b59f-ee92-443f-91ef-96ad5df79605.png">

Which is confusing for the user who don't know that the _old_ value is used. The mock also indicated that we should show `Unknown` instead. 

Implementation-wise it's a bit complicated since you decided to use `undefined` value as `total` and didn't left an option to store `Unknown`. That's why in the issue description I proposed to define it as  composite type `BigNumber | 'total'` (or rather `undefined | BigNumber | 'total'`). This of course is tricky since you will need to check the against the total string everywhere. Other (potentially better) option would be to set `value` to `NaN` during errors and then handle `NaN` in the parent as well.